R/palaeorotate.R

```diff
@@ -419,108 +419,130 @@ palaeorotate <- function(occdf, lng = "lng", lat = "lat", age = "age",
+                           coords[, age])
+    occdf$match <- paste0(occdf[, lng],
+                          occdf[, lat],
+                          occdf[, age])
+
+    # Prepare points query
+    # Split dataframe by age
+    coord_list <- split(x = coords, f = coords[, age])
+    # Split by chunk size
+    list_size <- lapply(coord_list, nrow)
+    subsplit <- names(which(list_size > chunks))
+    for (i in subsplit) {
+      tmp <- coord_list[[i]]
+      coord_list[[i]] <- split(x = tmp,
+                               f = ceiling(seq_len(nrow(tmp)) / chunks))
+    }
+
     # Run across models
-    for (m in model) {
+    multi_model <- lapply(model, function(m) {
       # Inform user which model is running
       message(m)
       # Run across unique ages
       rotations <- pblapply(X = uni_ages, function(i) {
         # Subset to age of interest
-        tmp <- coords[which(coords[, age] == i), ]
-        # How many rows?
-        nr <- nrow(tmp)
-        # Generate chunk bins
-        chk <- seq(from = 0, to = nr + chunks, by = chunks)
-        # Update final bin to equal nrow
-        chk[length(chk)] <- nr
-        # If ultimate chunk is the same as penultimate, drop
-        if (chk[length(chk)] == chk[length(chk) - 1]) {
-          chk <- chk[-length(chk)]
+        if (is.data.frame(coord_list[[as.character(i)]])) {
+          tmp <- coord_list[as.character(i)]
+        } else {
+          tmp <- coord_list[[as.character(i)]]
         }
-        # Create empty df
-        rot_df <- data.frame()
-        # Run across chunks
-        for (x in 2:length(chk)) {
-          # Lower index
-          ind_l <- chk[x - 1] + 1
-          # Upper index
-          ind_u <- chk[x]
-          # Generate API
-          tmp_chunk <- tmp[ind_l:ind_u, c(lng, lat, age)]
-          tmp_string <- toString(as.vector(t(tmp_chunk[, c(lng, lat)])))
-          api <- sprintf("?points=%s&time=%f&model=%s",
-                         gsub(" ", "", tmp_string), i, m)
-          api <- paste0("https://gws.gplates.org/reconstruct/",
-                        "reconstruct_points/",
-                        api, "&return_null_points")
-          # Call API
-          rots <- RETRY(verb = "GET",
-                        url = api,
-                        times = 5,
-                        pause_min = 1,
-                        pause_base = 1,
-                        pause_cap = 10)
-          # Extract coordinates
-          rots <- content(x = rots, as = "parsed")$coordinates
-          # Replace NULL values with NA
-          rpl <- which(rots == "NULL")
-          if (length(rpl) != 0) {
-            for (r in rpl) {
-              rots[[r]] <- list(NA, NA)
-            }
+        # Build requests
+        request <- lapply(tmp, function(x) {
+          req <- query[[m]]
+          req$time <- i
+          req$model <- m
+          pts <- paste(x[, lng], x[, lat], sep = ",")
+          pts <- toString(pts)
+          pts <- gsub(" ", "", pts)
```

Comment:
I think this can be replaced with just the following:
`paste(x[, lng], x[, lat], sep = ",", collapse = ",")`